matching-funds/app/daffy.server.ts

```diff
@@ -0,0 +1,88 @@
+import { z } from "zod";
+
+let BASE_URL = new URL("http://localhost:3000/");
+
+export class DaffyClient {
+  private token: string;
+  private type: "oauth" | "api-key";
+
+  constructor(token: string) {
+    this.token = token;
+    if (token.startsWith("daffy_production_")) {
+      this.type = "api-key";
+    } else {
+      this.type = "oauth";
+    }
```

Comment:
nit: plz use a ternary. also, i _really_ think that "daffy_" is sufficient

```suggestion
    this.type = token.startsWith("daffy_production_") ? "api-key" : "oauth";
```